src/ophyd_async/plan_stubs/_fly.py

```diff
@@ -128,18 +130,10 @@ def fly_and_collect(
 
     done = False
     while not done:
-        try:
-            yield from bps.wait(group=group, timeout=0.5)
-        except TimeoutError:
-            pass
-        else:
-            done = True
-        yield from bps.collect(
-            *detectors,
-            return_payload=False,
-            name=stream_name,
+        done = yield from bps.wait(
+            group=group, timeout=_SECONDS_TO_FORCE_COLLECT_AFTER, move_on=True
```

Comment:
We should add `flush_period: float = 0.5` to the function parameters, then use that rather than a global
```suggestion
            group=group, timeout=flush_period, move_on=True
```